### ISSUE TITLE
API: jl_get_julia_bindir, julia_bin, image_file exported but not declared

### ISSUE THREAD

**Issue by alhirzel** [JuliaLang/julia#29558]
The functions jl_get_julia_bindir, jl_get_julia_bin, jl_get_image_file are not declared in julia.h or julia_threading.h, even though they are exported in jlapi.c. This is in the same spirit as #29556, but without an API nit-pick.

**Comment by yuyichao** [comment]
Not all exported functions are C API. A lot are only called from julia.

**Comment by alhirzel** [comment]
Reject if appropriate :)

Is there is a canonical list of C API functions?

**Comment by yuyichao** [comment]
The headers

**Comment by fingolfin** [comment]
I agree that a more stable and comprehensive kernel API spec would be nice, but at it is, the only relevant thing are indeed the headers, and even there changes will happen going from Julia 1.x to 1.(x+1). It would be interesting to work towards specifying a "core API" that has stronger stability guarantees. But this requires someone who would spearhead this.
